### PR TITLE
added systemd unit file and target for make

### PR DESCRIPTION
**Pull request by aprasolova**
#326

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,6 +19,21 @@ build: deps
 		&& export CGO_ENABLED=0 \
 		&& go build -v -mod=vendor -ldflags $(BUILD_FLAGS) -o ${BINARY} ./cli/main.go
 
+install: build
+	@echo "=> Preparing directories and configs"
+	@id neo-go || useradd -s /usr/sbin/nologin -d /var/lib/neo-go neo-go \
```
Comment (roman-khimov) [inline Makefile]: Please leave this setup to RPM/DEB postinstall scripts (hmm, we're missing issues for these now).

File: Makefile [hunk 2]
```diff
@@ -19,6 +19,21 @@ build: deps
...
+		&& mkdir -p /var/lib/neo-go \
+		&& chown -R neo-go:neo-go /var/lib/neo-go
+	@echo "=> Installing systemd service"
+	@mkdir -p /etc/neo-go \
```
Comment (roman-khimov) [inline Makefile]: Please use `$(DESTDIR)` to be able to override paths, like
```
mkdir -p $(DESTDIR)/etc/neo-go
```
Or even better:
```
mkdir -p $(DESTDIR)$(sysconfdir)/neo-go
```
Same thing for `$(bindir)` and `$(systemd_unitdir)`. You'll need to add defaults to the `Makefile`, but in the end all of these things will be overridable and every package maintainer will just love you for that.

File: Makefile [hunk 3]
```diff
@@ -19,6 +19,21 @@ build: deps
...
+		&& mkdir -p /etc/systemd/system/ \
+		&& cp ./neo-go.service /etc/systemd/system/ \
+		&& cp ./config/protocol.mainnet.yml /etc/neo-go \
+		&& systemctl enable neo-go.service \
```
Comment (roman-khimov) [inline Makefile]: This is a postinst part.

File: Makefile [hunk 4]
```diff
@@ -19,6 +19,21 @@ build: deps
...
 
+install: build
+	@echo "=> Preparing directories and configs"
+	@id neo-go || useradd -s /usr/sbin/nologin -d /var/lib/neo-go neo-go \
+		&& mkdir -p /var/lib/neo-go \
+		&& chown -R neo-go:neo-go /var/lib/neo-go
+	@echo "=> Installing systemd service"
+	@mkdir -p /etc/neo-go \
+		&& mkdir -p /etc/systemd/system/ \
+		&& cp ./neo-go.service /etc/systemd/system/ \
```
Comment (roman-khimov) [inline Makefile]: I think the default should be in the `/usr/lib`, `/etc` is for the admin.

File: Makefile [hunk 5]
```diff
@@ -19,6 +19,21 @@ build: deps
...
+		&& cp ./config/protocol.mainnet.yml /etc/neo-go \
```
Comment (roman-khimov) [inline Makefile]: Please also install configs for `testnet` and `privnet`.

File: Makefile [hunk 6]
```diff
@@ -19,6 +19,21 @@ build: deps
...
+		&& systemctl enable neo-go.service \
+		&& mv ./bin/neo-go /usr/local/bin/neo-go \
+		&& chown neo-go:neo-go /usr/local/bin/neo-go
```
Comment (roman-khimov) [inline Makefile]: Also a `postinst` thing.

File: Makefile [hunk 7]
```diff
@@ -19,6 +19,21 @@ build: deps
...
+	@id neo-go || useradd -s /usr/sbin/nologin -d /var/lib/neo-go neo-go \
+		&& mkdir -p /var/lib/neo-go \
+		&& chown -R neo-go:neo-go /var/lib/neo-go
+	@echo "=> Installing systemd service"
+	@mkdir -p /etc/neo-go \
+		&& mkdir -p /etc/systemd/system/ \
+		&& cp ./neo-go.service /etc/systemd/system/ \
+		&& cp ./config/protocol.mainnet.yml /etc/neo-go \
+		&& systemctl enable neo-go.service \
+		&& mv ./bin/neo-go /usr/local/bin/neo-go \
```
Comment (roman-khimov) [inline Makefile]: `mv` is wrong here, please use `install -m 0755`.

Comment (roman-khimov) [reply Makefile]: Oh, and also please use `/usr/bin` path here.

File: neo-go.service [hunk 8]
```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=Neo Go node
+Requires=network.target
+
+[Service]
+Type=simple
+ExecStart=/usr/local/bin/neo-go node --mainnet --config-path=/etc/neo-go
```
Comment (roman-khimov) [inline neo-go.service]: `/usr/bin`.

File: Makefile (hunk 5 above)
Comment (aprasolova) [reply Makefile]: done

File: Makefile (hunk 1 above)
Comment (aprasolova) [reply Makefile]: done

File: Makefile (hunk 2 above)
Comment (aprasolova) [reply Makefile]: done

File: Makefile (hunk 4 above)
Comment (aprasolova) [reply Makefile]: Moved to `/lib/systemd/system`, this path is considered for custom units.

File: Makefile (hunk 7 above)
Comment (aprasolova) [reply Makefile]: done

File: neo-go.service (hunk 8 above)
Comment (aprasolova) [reply neo-go.service]: done

**Comment by aprasolova** [comment]
I added `postinst` target. It doesn't look complete and sufficient, but it contains some [non-obvious] actions required for service to run.

File: neo-go.service [hunk 9]
```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=Neo Go node
+Requires=network.target
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/neo-go node --mainnet --config-path=/etc/neo-go
```
Comment (roman-khimov) [inline neo-go.service]: I think we can we make it a little better by templating this service file and expanding paths to what is specified in the `Makefile`. You can write something like
```
ExecStart=$$BINDIR$$/neo-go node --mainnet --config-path=/etc/neo-go
```
here and then in the Makefile do
```
neo-go.service: neo-go.service.template
    sed -r 's,\$\$BINDIR\$\$,$(BINDIR)' $< >$@

install: build neo-go.service
...
```
The same applies to `config-path` and `WorkingDirectory`.

Comment (aprasolova) [reply neo-go.service]: done

**Pull request merged**